epochStart/metachain/systemSCs_test.go

```diff
@@ -956,21 +961,119 @@ func createFullArgumentsForSystemSCProcessing(enableEpochsConfig config.EnableEp
 func TestSystemSCProcessor_ProcessSystemSmartContractInitDelegationMgr(t *testing.T) {
 	t.Parallel()
 
-	args, _ := createFullArgumentsForSystemSCProcessing(config.EnableEpochs{
-		StakingV2EnableEpoch: 1000,
-	}, testscommon.CreateMemUnit())
-	s, _ := NewSystemSCProcessor(args)
-
-	validatorsInfo := state.NewShardValidatorsInfoMap()
-	err := s.ProcessSystemSmartContract(validatorsInfo, &block.Header{})
-	assert.Nil(t, err)
+	t.Run("flag not active - activation epoch is in the future", func(t *testing.T) {
+		args, _ := createFullArgumentsForSystemSCProcessing(config.EnableEpochs{
+			DelegationSmartContractEnableEpoch: 39,
+			StakingV2EnableEpoch:               1000,
+		}, testscommon.CreateMemUnit())
+		s, _ := NewSystemSCProcessor(args)
+		handler := s.enableEpochsHandler.(enableEpochHandlerWithEpochConfirm)
+		handler.EpochConfirmed(37, 0)
```

Comment:
I think there is no need for this, you could simply call `s.EpochConfirmed` and also get rid of the interface definition

Reply:
refactored completely